### PR TITLE
Refactor phoenix.js

### PR DESCRIPTION
**Pull request by webdeb**
Use ES6 where possible, rely on wide used standards, use let only for variables, which are going to be changed. Otherwise use `const` and so on. 

I hope it provides also better readability.

### CODE REVIEW

File: assets/js/phoenix.js [hunk 1]
```diff
@@ -255,34 +264,37 @@ class Push {
...
-
   // private
-
-  matchReceive({status, response, ref}){
-    this.recHooks.filter( h => h.status === status )
-                 .forEach( h => h.callback(response) )
+  matchReceive({ status, response, ref }) {
+    this.recHooks.forEach(h =>
+      h.status === status && h.callback(response)
+    )
```
Comment (webdeb) [inline assets/js/phoenix.js]: Remove unnecessary filter loop

**Comment by chrismccord** [comment]
We don't accept purely stylistic PRs as it introduces churn and potential conflicts for progress going forward. Thanks!

**Comment by aphillipo** [comment]
This a little more efficient (o(n) vs o(2n)) as well as being a stylistic change. It’s probably not necessary to loop twice here ever that I can think of. If recHooks can never be large then probably it’s not worth changing but if it can be a few hundred items maybe consider it.

**Comment by webdeb** [comment]
The whole `phoenix.js` feels a bit messy, and inconsistent in its styling, this reduces readability IMHO.

However, #2368 removes the unnecessary double-loops.

**Comment by odyright** [comment]
Hello, consider if we use eslint, what would be the best config...?

**Comment by webdeb** [comment]
@odyright I would stick with the **eslint:recommended**, then add your own as you go.

**Pull request closed without merging**